### ISSUE TITLE
Allow Xcode paths with different patterns

### ISSUE THREAD

**Issue by bc-lee** [maxim-lobanov/setup-xcode#17]
First of all, thank you for the great GH action.

From https://github.com/maxim-lobanov/setup-xcode/blob/f3914cfd8e176f63fe6f27879260510b0bd482fc/src/xcode-utils.ts#L29

it accepts Xcode paths like `/Application/Xcode_12.4.app` , but not `/Application/Xcode12.4.app` .

I think both type of paths are popular for maintaining multiple Xcode versions.

It would be nice to accept the second kind of paths. Thank you.

**Comment by dentuzhik** [comment]
Considering that right now pList is used for parsing, I think it would be rather safe to make the whole sequence of `_([\d.]+)(_beta)?` optional

**Comment by bc-lee** [comment]
Yes, the pList file parsing seems to work correctly, so we can simplify that regex a lot.

**Comment by maxim-lobanov** [comment]
Probably `Xcode[\w\.]+\.app` should be enough. Or even `^Xcode.+\.app$`

Just to clarify, you are needed it for self-hosted machines, right? Because as far as I know, Hosted machines follow the current pattern.

**Comment by bc-lee** [comment]
Yes. I would like to use this in a self-hosted runner with multiple versions of Xcode already installed, such as `/Application/Xcode12.4.app`.

**Comment by dentuzhik** [comment]
In my fork I ended up with following regexp to allow recognising `Xcode.app` too, which is a default name if you install from AppStore:
`/Xcode(_([\d.]+))?(_beta)?\.app/`

**Comment by maxim-lobanov** [comment]
Prepared small PR with fix: https://github.com/maxim-lobanov/setup-xcode/pull/19
Validated that action continue to work as expected on Hosted runners.

@bc-lee , would be appreciate for any testing on self-hosted.
You can test PR branch via 
```yml
steps:
- uses: maxim-lobanov/setup-xcode@improve-xcode-regex
  with:
    xcode-version: '<version>'
```

**Comment by bc-lee** [comment]
It works perfectly! Thank you!

**Comment by maxim-lobanov** [comment]
I have released `v1.2.2` with this fix and also updated `v1` to point v1.2.2

**Comment by ffittschen** [comment]
Hi @maxim-lobanov, I just tried to use `v1.2.2` that includes this change, but it could not find any Xcode versions. Upon investigating the issue, I saw that the `v1.2.2` contains only the source changes, but the `dist/index.js` was not updated. Could you please create a new release that includes these changes in `dist/index.js` as well? 🙂

**Comment by maxim-lobanov** [comment]
@ffittschen , thank you for heads up! I have missed it.
Released `v1.2.3` with rebuilt index.js and updated `v1`

**Comment by ffittschen** [comment]
Perfect, thanks a lot!
